server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -273,6 +280,61 @@ private long getDenseVectorValueCount(final LeafReader atomicReader) throws IOEx
         return count;
     }
 
+    /**
+     * Returns the {@link SparseVectorStats} for this engine
+     */
+    public SparseVectorStats sparseVectorStats(MappingLookup mappingLookup) {
+        try (Searcher searcher = acquireSearcher(DOC_STATS_SOURCE, SearcherScope.INTERNAL)) {
+            return sparseVectorStats(searcher.getIndexReader(), mappingLookup);
+        }
+    }
+
+    protected final SparseVectorStats sparseVectorStats(IndexReader indexReader, MappingLookup mappingLookup) {
+        long valueCount = 0;
+
+        // we don't wait for a pending refreshes here since it's a stats call instead we mark it as accessed only which will cause
+        // the next scheduled refresh to go through and refresh the stats as well
+        for (LeafReaderContext readerContext : indexReader.leaves()) {
+            try {
+                valueCount += getSparseVectorValueCount(readerContext.reader(), mappingLookup);
+            } catch (IOException e) {
+                logger.trace(() -> "failed to get sparse vector stats for [" + readerContext + "]", e);
+            }
+        }
+        return new SparseVectorStats(valueCount);
+    }
+
+    private long getSparseVectorValueCount(final LeafReader atomicReader, MappingLookup mappingLookup) throws IOException {
+        long count = 0;
+
+        if (mappingLookup == null) {
```

Comment:
The check could be done in `sparseVectorStats` already to avoid accessing the leaves?